### PR TITLE
Feature/single news ui

### PR DESCRIPTION
**Pull request by RomanDeveloperAcc**
Created new page with the single news item UI.

### CODE REVIEW

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css [hunk 1]
```diff
@@ -0,0 +1,102 @@
+* {
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+    font-family: 'Montserrat', sans-serif;
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css]: General: according to mock for title you should use font family 'Montserrat' and for another elements 'OpenSans'. Please double check it with mockup and fix. And remove *, it's bad practice

Comment (RomanDeveloperAcc) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css]: Fixed

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    line-height: 19px;
+    text-decoration-line: underline;
+}
+
+.button-link {
+    color: #494A49;
+}
+
+.button-arrow {
+  margin-right: 4px;
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css]: bad alignment

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    padding: 0 292px 48px;
+    justify-content: center;
+}
+
+.news-info-dot {
+    padding: 0 32px 0;
+}
+
+.news-links-images {
+  flex-direction: column;
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css]: bad alignment

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  margin-top: 37px;
+}
+
+.news-links-img {
+  margin: 9px 0;
+}
+
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css]: the same as line 85

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+.news-text-container {
+  margin: 0 98px;
+  display: flex;
+  justify-content: space-between;
+}
+
+.news-text {
+    padding: 48px 97px 0;
+    text-align: left;
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css]: general: please check alignment in styles

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts [hunk 6]
```diff
@@ -1,15 +1,11 @@
 import { Component, OnInit } from '@angular/core';
 
 @Component({
-  selector: 'app-eco-news-detail',
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: Why did you delete selector?

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css (hunk 4 above)
Comment (RomanDeveloperAcc) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css]: fixed

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css (hunk 5 above)
Comment (RomanDeveloperAcc) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.css]: fixed

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts (hunk 6 above)
Comment (RomanDeveloperAcc) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: fixed

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html [hunk 7]
```diff
@@ -1 +1,63 @@
-<p>eco-news-detail works!</p>
+<div class="container">
+  <div class="top-elements">
+    <div class="back-button">
+      <a [routerLink]="['/news']"
+         class="button-link">
+        <div class="button-content">
+          <div class="button-arrow">
+            <img src="../../../../assets/img/icon/econews/arrow_left.svg"
```
Comment (KimAT25) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: Move pictures in variable in extra file and use binding

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html [hunk 8]
```diff
@@ -1 +1,63 @@
...
+    <div class="tags-item">Events</div>
+    <div class="tags-item">News</div>
+    <div class="tags-item">Initiatives</div>
+  </div>
+  <div class="news-content">
+    <div class="news-title">{{ newsItem.title }}</div>
+    <div class="news-info">
+      <div class="news-info-date">{{ newsItem.creationDate }}</div>
+      <div class="news-info-dot">
+        <img src="../../../../assets/img/icon/econews/ellipse.svg"
```
Comment (KimAT25) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: same as line 8

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html [hunk 9]
```diff
@@ -1 +1,63 @@
...
+             alt=""
+        >
+      </div>
+      <div class="news-info-author">by {{ newsItem.name }}</div>
+    </div>
+    <div class="news-image">
+      <img src="../../../../assets/img/icon/econews/news-default-large.png"
```
Comment (KimAT25) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: same as line 8

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html [hunk 10]
```diff
@@ -1 +1,63 @@
...
+           alt=""
+           class="news-image-img"
+      >
+    </div>
+    <div class="news-text-container">
+      <div class="news-links-images">
+        <img src="../../../../assets/img/icon/econews/twitter.svg"
```
Comment (KimAT25) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: same as line 8

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html [hunk 11]
```diff
@@ -1 +1,63 @@
...
+             alt=""
+             class="news-links-img"
+        >
+        <img src="../../../../assets/img/icon/econews/linkedin.svg"
```
Comment (KimAT25) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: same as line 8

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html [hunk 12]
```diff
@@ -1 +1,63 @@
...
+             alt=""
+             class="news-links-img"
+        >
+        <img src="../../../../assets/img/icon/econews/facebook.svg"
```
Comment (KimAT25) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: same as line 8

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html [hunk 13]
```diff
@@ -1 +1,63 @@
-<p>eco-news-detail works!</p>
+<div class="container">
+  <div class="top-elements">
+    <div class="back-button">
+      <a [routerLink]="['/news']"
+         class="button-link">
+        <div class="button-content">
+          <div class="button-arrow">
+            <img src="../../../../assets/img/icon/econews/arrow_left.svg"
+                 alt=""
```
Comment (KimAT25) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: add alt text

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html [hunk 14]
```diff
@@ -1 +1,63 @@
...
+    <div class="tags-item">News</div>
+    <div class="tags-item">Initiatives</div>
+  </div>
+  <div class="news-content">
+    <div class="news-title">{{ newsItem.title }}</div>
+    <div class="news-info">
+      <div class="news-info-date">{{ newsItem.creationDate }}</div>
+      <div class="news-info-dot">
+        <img src="../../../../assets/img/icon/econews/ellipse.svg"
+             alt=""
```
Comment (KimAT25) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: same as line 9

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html [hunk 15]
```diff
@@ -1 +1,63 @@
...
+        >
+      </div>
+      <div class="news-info-author">by {{ newsItem.name }}</div>
+    </div>
+    <div class="news-image">
+      <img src="../../../../assets/img/icon/econews/news-default-large.png"
+           alt=""
```
Comment (KimAT25) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: same as line 9

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html [hunk 16]
```diff
@@ -1 +1,63 @@
...
+           class="news-image-img"
+      >
+    </div>
+    <div class="news-text-container">
+      <div class="news-links-images">
+        <img src="../../../../assets/img/icon/econews/twitter.svg"
+             alt=""
```
Comment (KimAT25) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: same as line 9

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html [hunk 17]
```diff
@@ -1 +1,63 @@
...
+             class="news-links-img"
+        >
+        <img src="../../../../assets/img/icon/econews/linkedin.svg"
+             alt=""
```
Comment (KimAT25) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: same as line 9

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html [hunk 18]
```diff
@@ -1 +1,63 @@
...
+             class="news-links-img"
+        >
+        <img src="../../../../assets/img/icon/econews/facebook.svg"
+             alt=""
```
Comment (KimAT25) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: same as line 9

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html (hunk 7 above)
Comment (nabermi2020) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: not required

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html (hunk 8 above)
Comment (nabermi2020) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: not required

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html (hunk 9 above)
Comment (nabermi2020) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: not required

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html (hunk 10 above)
Comment (nabermi2020) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: not required

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html (hunk 11 above)
Comment (nabermi2020) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: not required

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html (hunk 15 above)
Comment (nabermi2020) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: not required

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html (hunk 16 above)
Comment (nabermi2020) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.html]: not required

File: src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts [hunk 19]
```diff
@@ -1,15 +1,38 @@
-import { Component, OnInit } from '@angular/core';
```
Comment (nabermi2020) [inline src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: please, fix errors in the console

Comment (nabermi2020) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: image

Comment (nabermi2020) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: interpolation is incorrect

Comment (nabermi2020) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: date format is invalid
Please use date pipe

Comment (KimAT25) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: I agreed with Roman, that I will fix issue with interpolation, as we need to change model file.

Comment (KimAT25) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: I agreed with Roman, that I will fix issue with interpolation, as Model file have to be changed according to new JSON file.

Comment (KimAT25) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: For Date pipe use same example of format date as on the Figma mockup. We agreed this today with BA and Designer.

Comment (RomanDeveloperAcc) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: > For Date pipe use same example, of format date, as on the Figma mockup. We agreed this today with BA and Designer.

I will add pipe in my html

Comment (RomanDeveloperAcc) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: > date format is invalid
> Please use date pipe('longDate')

It is better to use mediumDate due to mockup

Comment (KimAT25) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: use simple date, as on Mockup no time only date

Comment (RomanDeveloperAcc) [reply src/app/component/general/eco-news/eco-news-detail/eco-news-detail.component.ts]: > please, fix errors in the console

fixed
